RecoHI/HiEvtPlaneAlgos/src/HiEvtPlaneFlatProducer.cc

```diff
@@ -110,21 +155,50 @@ typedef TrackingParticleRefVector::iterator               tp_iterator;
 //
 // constructors and destructor
 //
-HiEvtPlaneFlatProducer::HiEvtPlaneFlatProducer(const edm::ParameterSet& iConfig)
+HiEvtPlaneFlatProducer::HiEvtPlaneFlatProducer(const edm::ParameterSet& iConfig):runno_(0)
 {
+  UseEtHF = kFALSE;
+  nCentBins_ = 200.;
+  
+  centralityVariable_ = iConfig.getParameter<std::string>("centralityVariable");
+  if(iConfig.exists("nonDefaultGlauberModel")){
+    centralityMC_ = iConfig.getParameter<std::string>("nonDefaultGlauberModel");
+  }
+  centralityLabel_ = centralityVariable_+centralityMC_;
 
-  vtxCollection_  = consumes<reco::VertexCollection>(iConfig.getParameter<edm::InputTag>("vtxCollection_"));
-  inputPlanes_ = consumes<reco::EvtPlaneCollection>(iConfig.getParameter<edm::InputTag>("inputPlanes_"));
-  centrality_ = consumes<int>(iConfig.getParameter<edm::InputTag>("centrality_"));
-  storeNames_ = 1;
+  centralityBinTag_ = iConfig.getParameter<edm::InputTag>("centralityBinTag_");
+  centralityBinToken = consumes<int>(centralityBinTag_);
+
+  centralityTag_ = iConfig.getParameter<edm::InputTag>("centralityTag_");
+  centralityToken = consumes<reco::Centrality>(centralityTag_);
+
+  vertexTag_  = iConfig.getParameter<edm::InputTag>("vertexTag_");
+  vertexToken = consumes<std::vector<reco::Vertex>>(vertexTag_);
+
+  trackTag_ = iConfig.getParameter<edm::InputTag>("trackTag_");
+  trackToken = consumes<reco::TrackCollection>(trackTag_);
+
+  inputPlanesTag_ = iConfig.getParameter<edm::InputTag>("inputPlanesTag_");
+  inputPlanesToken = consumes<reco::EvtPlaneCollection>(inputPlanesTag_);
+
+  FlatOrder_ = iConfig.getUntrackedParameter<int>("FlatOrder_", 9);
```

Comment:
- make all parameters affecting physics results tracked
- remove the trailing underscore in they python parameters string names